tests/mpmc_channel.rs

```diff
@@ -918,6 +941,41 @@ macro_rules! gen_mpmc_tests {
                 assert_send_done(cx, &mut send_fut2, Ok(()));
                 assert_send_done(cx, &mut send_fut3, Ok(()));
             }
+
+            #[test]
+            fn stream_smoke_test() {
+                let channel = ChannelType::new();
```

Comment:
This just checks the non-shared variant. We should also have a test for the shared version.